tests/test_databrowser_restapi.py

```diff
@@ -0,0 +1,364 @@
+
+    res6 = requests.get(
+        f"{test_server}/api/databrowser/metadata_search/cmip6/uri",
+        params={"facets": "activity_id"},
+    ).json()
+    assert len(res6["facets"].keys()) == 1
+
+
+def test_contains_not(test_server: str) -> None:
+    """Test for searches that should *not* contain values."""
+
+    res1 = requests.get(
+        f"{test_server}/api/databrowser/metadata_search/freva/file",
+        params={
+            "dataset": ["-cmip6-swift", "not cmip6-fs"],
+            "project": "cmip6",
+        },
+    ).json()
+    assert "cmip6-swift" not in res1["facets"]["dataset"]
+    assert "cmip6-fs" not in res1["facets"]["dataset"]
+    assert "cmip6-hsm" in res1["facets"]["dataset"]
+    res2 = requests.get(
+        f"{test_server}/api/databrowser/metadata_search/freva/file",
+        params={
+            "project_not_": "cmip6",
+        },
+    ).json()
+    assert "cmip6" not in res2["facets"]["project"]
+
+
+def test_intake_search(test_server: str) -> None:
+    """Test the creation of intake catalogues."""
+    res1 = requests.get(
+        f"{test_server}/api/databrowser/intake_catalogue/cmip6/uri",
+        params={"activity_id": "cmip", "multi-version": True},
+    )
+    assert res1.json() == json.loads(res1.text)
+    res2 = requests.get(
+        f"{test_server}/api/databrowser/intake_catalogue/cmip6/uri",
+    )
+    assert len(res2.json()["catalog_dict"]) > len(res1.json()["catalog_dict"])
+    res3 = requests.get(
+        f"{test_server}/api/databrowser/intake_catalogue/cmip6/uri",
+        params={
+            "activity_id": "cmip",
+            "multi-version": False,
+        },
+    )
+    assert len(res1.json()["catalog_dict"]) > len(res3.json()["catalog_dict"])
+    res4 = requests.get(
+        f"{test_server}/api/databrowser/intake_catalogue/cmip6/uri",
+        params={
+            "multi-version": False,
+            "max-results": 1,
+        },
+    )
+    assert res4.status_code == 413
+
+
+def test_bad_intake_request(test_server: str) -> None:
+    """Test for a wrong intake request."""
+    res1 = requests.get(
+        f"{test_server}/api/databrowser/intake_catalogue/cmip6/uri",
+        params={"activity_id": "cmip2"},
+    )
+    assert res1.status_code == 404
+
+
+def test_parameter_validation(test_server: str) -> None:
+    """Test if only valid parameter requests make it."""
+
+    res1 = requests.get(
+        f"{test_server}/api/databrowser/data_search/cmip6/uri",
+        params={"activity_id": "cmip", "translate": "false"},
+    ).status_code
+    res2 = requests.get(
+        f"{test_server}/api/databrowser/data_search/cmip6/uri",
+        params={"product": "cmip", "translate": "true"},
+    ).status_code
+    res3 = requests.get(
+        f"{test_server}/api/databrowser/data_search/cmip6/uri",
+        params={"activity_": "cmip"},
+    ).status_code
+    assert res1 == res2 == res3 == 422
+
+
+def test_no_mongo_parameter_insert(test_server: str) -> None:
```

Comment:
I think for making the coverage full, we need a test like this for `_delete_from_mongo`. i just copied your `test_no_mongo_parameter_insert` and made it based on that. 

```python
def test_no_mongo_parameter_delete(test_server: str) -> None:
    """Test the failed deletion of data into the mongodb."""
    from freva_rest.rest import server_config
    token = auth["access_token"]
    with MongoClient(server_config.mongo_url) as mongo_client:
        collection = mongo_client[server_config.mongo_db]["user_data"]
        collection.drop()
        assert collection.find_one() is None
    env = os.environ.copy()
    env["MONGO_HOST"] = "foo.bar.de"
    with mock.patch.dict(os.environ, env, clear=True):
        server_config.power_cycle_mongodb()
        res1 = requests.delete(
            f"{test_server}/api/databrowser/userdata",
            json={},
            headers={"Authorization": f"Bearer {auth['access_token']}"},
        ).status_code
        assert response.status_code == 202
    server_config.power_cycle_mongodb()
    with MongoClient(server_config.mongo_url) as mongo_client:
        collection = mongo_client[server_config.mongo_db]["user_data"]
        assert collection.find_one() is None
```

Reply:
I've taken the liberty to make converagerc not to consider those lines because we don't do anything special other than logging an error if an issue occurs.